src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs

```diff
@@ -145,7 +145,14 @@ private protected virtual void CheckLabelCompatible(SchemaShape.Column labelCol)
             IDataView validationSet = null, IPredictor initPredictor = null)
         {
             var trainRoleMapped = MakeRoles(trainSet);
-            var validRoleMapped = validationSet == null ? null : MakeRoles(validationSet);
+            CheckInputSchema(SchemaShape.Create(trainSet.Schema));
```

Comment:
>CheckInputSchema [](start = 12, length = 16)

Nit: Ordering for `train` and `valid` in `Make` and `Check` are different. I'd prefer to have the same sequence.